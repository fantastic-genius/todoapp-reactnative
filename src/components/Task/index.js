import React from 'react';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';
import { Ionicons, MaterialCommunityIcons, MaterialIcons } from '@expo/vector-icons'

const Task = ({ task, checked, handleSelect }) => {

  return (
    <View style={styles.taskCon}>
      <TouchableOpacity 
        onPress={handleSelect}
      >
        {checked ? (
          <Ionicons name='ios-checkbox' size={24} color='#2065ff' />
        ) : (
          <MaterialIcons name="check-box-outline-blank" size={24} color="black" />
        )}
      </TouchableOpacity>
      <Text style={styles.itemText}>{task.name}</Text>
      <Text style={styles.itemText}>{task.due_date}</Text>
      {task.status ===  'done' ? (
        <MaterialCommunityIcons name='progress-check' size={24} color='green' />
      ) : task.status === 'in-progress' ? (
        <MaterialCommunityIcons name='progress-clock' size={24} color='orange' />
      ) : (
        <MaterialCommunityIcons name='progress-alert' size={24} color='red' />
      )}
    </View>
  )
}

const styles = StyleSheet.create({
  taskCon: {
    backgroundColor: '#ffffff',
    height: 50,
    flexDirection: 'row',
    alignItems: 'center',
    paddingHorizontal: 15,
    marginBottom: 15
  },
  itemText: {
    flex: 1,
    color: '#717171',
    fontSize: 13,
    paddingHorizontal: 4
  }
});

export default Task;
