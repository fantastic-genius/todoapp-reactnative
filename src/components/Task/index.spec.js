import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'
import Task from './index';

describe('<Task />', () => {
  const task = {
    id: '1',
    name: 'Task 1',
    due_date: '9:00AM 12-1-2019', 
    status: 'done'
  }
  it('Has four children', async () => {
    const component = renderer.create(<Task task={task} />).toJSON();
    expect(component.children.length).toBe(4);
  });
  
  it('renders correctly', async () => {
    const component = renderer.create(<Task task={task} />).toJSON();
    expect(component).toMatchSnapshot();
  });

});

