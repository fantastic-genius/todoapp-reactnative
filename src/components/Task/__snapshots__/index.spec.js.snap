// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Task /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flexDirection": "row",
      "height": 50,
      "marginBottom": 15,
      "paddingHorizontal": 15,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text />
  </View>
  <Text
    style={
      Object {
        "color": "#717171",
        "flex": 1,
        "fontSize": 13,
        "paddingHorizontal": 4,
      }
    }
  >
    Task 1
  </Text>
  <Text
    style={
      Object {
        "color": "#717171",
        "flex": 1,
        "fontSize": 13,
        "paddingHorizontal": 4,
      }
    }
  >
    9:00AM 12-1-2019
  </Text>
  <Text />
</View>
`;
